browser/brave_shields/ad_block_service_browsertest.cc

```diff
@@ -2251,6 +2251,94 @@ IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringCustomStyle) {
   EXPECT_EQ(base::Value(true), result.value);
 }
 
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringRemoveStatic) {
+  UpdateAdBlockInstanceWithRules(
+      "###ad-banner:remove()\n"
+      "##.whatever:remove()");
+
+  GURL tab_url =
+      embedded_test_server()->GetURL("b.com", "/cosmetic_filtering.html");
+  NavigateToURL(tab_url);
+
+  content::WebContents* contents = web_contents();
+  
+  EXPECT_EQ(true, EvalJs(contents,
+                         "check('#ad-banner', existence(false))"));
+}
+
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringRemoveDynamic) {
+  UpdateAdBlockInstanceWithRules(
+      "##.blockme:remove()");
+
+  GURL tab_url =
+      embedded_test_server()->GetURL("b.com", "/cosmetic_filtering.html");
+  NavigateToURL(tab_url);
+
+  content::WebContents* contents = web_contents();
+
+  auto result =
+      EvalJs(contents,
+             "addElementsDynamically();\n"
+             "wait('.dontblockme', existence(true)).then(() =>"
+             "wait('.blockme', existence(false)))");
+  ASSERT_TRUE(result.error.empty());
+  EXPECT_EQ(base::Value(true), result.value);
+}
+
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringRemoveAttribute) {
+  UpdateAdBlockInstanceWithRules(
+      "##.ad img:remove-attr(src)\n"
+      "##img:remove-attr(whatever)");
+
+  GURL tab_url =
+      embedded_test_server()->GetURL("b.com", "/cosmetic_filtering.html");
+  NavigateToURL(tab_url);
+
+  content::WebContents* contents = web_contents();
+
+  
+  EXPECT_EQ(true, EvalJs(contents,
+                         "check('.ad img', attributes(['alt']))"));
+
+  // Sanity check selector
+  EXPECT_EQ(true, EvalJs(contents,
+                         "check('.relative-url-div img', attributes(['src']))"));
+}
+
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringRemoveAttributeDynamic) {
+  UpdateAdBlockInstanceWithRules(
+      "##img.blockme:remove-attr(src)");
+
+  GURL tab_url =
+      embedded_test_server()->GetURL("b.com", "/cosmetic_filtering.html");
+  NavigateToURL(tab_url);
+
+  content::WebContents* contents = web_contents();
+
+  auto result = EvalJs(contents,
+                       "addElementsDynamically();\n"
+                       "wait('img.blockme', attributes(['class']))");
+  ASSERT_TRUE(result.error.empty());
+  EXPECT_EQ(base::Value(true), result.value);
+}
+
+IN_PROC_BROWSER_TEST_F(AdBlockServiceTest, CosmeticFilteringRemoveClass) {
```

Comment:
Could we also check that remove logic works **per domain**?
I assume the common usage is something like `b.com##.ad:remove-class(ghi)`.
@antonok-edm WDYT?

Reply:
I have found that currently the tests are only working for `b.com##.ad:remove-class(ghi)` (I'm about to push an update to fix these broken tests by adding `b.com` to the start).

The tests don't actually pass without a domain.

Reply:
@atuchin-m looks like it is working as intended; cosmetic filters with an action (`:style`, `:remove`, `:remove-class`, `:remove-attr`) are only valid when a domain is specified.